service/kvmclusterconfig/v1/resource/encryptionkey/create.go

```diff
@@ -11,7 +11,19 @@ import (
 // Patch provided by NewUpdatePatch or NewDeletePatch. It creates k8s secret
 // for encryption key if needed.
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
-	return nil
+	secret, err := toSecret(createChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if secret != nil {
+		_, err = r.k8sClient.Core().Secrets(v1.NamespaceDefault).Create(secret)
+		if err != nil {
+			err = microerror.Mask(err)
+		}
+	}
```

Comment:
There's some standard logging missing: https://github.com/giantswarm/azure-operator/blob/fd4df26bee65aa24d8d3015864531de766d362f7/service/resource/resourcegroup/resource.go#L191-L193

This is the reason why we want checking nil in here.